ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -262,12 +263,26 @@ public BeaconState deserializeBeaconState(final Bytes serializedState) {
         .sszDeserialize(serializedState);
   }
 
-  public SignedBeaconBlock deserializeSignedBeaconBlock(final Bytes serializedState) {
-    final UInt64 slot = BeaconBlockInvariants.extractSignedBeaconBlockSlot(serializedState);
+  public SignedBeaconBlock deserializeSignedBeaconBlock(final Bytes serializedBlock) {
+    final UInt64 slot = BeaconBlockInvariants.extractSignedBeaconBlockSlot(serializedBlock);
     return atSlot(slot)
         .getSchemaDefinitions()
         .getSignedBeaconBlockSchema()
-        .sszDeserialize(serializedState);
+        .sszDeserialize(serializedBlock);
+  }
+
+  public ExecutionPayload deserializeExecutionPayload(final Bytes serializedPayload) {
+    final SpecVersion specVersion = forMilestone(SpecMilestone.BELLATRIX);
+    if (specVersion == null) {
+      throw new IllegalStateException("Bellatrix must be scheduled to decode execution payloads");
+    }
```

Comment:
This is possible, just putting it out there that it's not 100% ideal if ExecutionPayload changes on a hard fork...

Reply:
I think we can count on ExecutionPayload changing and need to design for it now, so can't just hard code the milestone to use.

Either need to lookup the milestone via the block's slot or store the milestone with the serialised payload. Ultimately we'll want to retrieve the payload from the EL rather than the DB so will have to have the block info available to do that.